### ISSUE TITLE
`no-duplicate-selectors` false positives with Less syntax

### ISSUE THREAD

**Issue by afc163** [stylelint/stylelint#6110]
### What steps are needed to reproduce the bug?

```jsx
git clone git@github.com:ant-design/ant-design.git
npm install
npm run lint:style
```

### What Stylelint configuration is needed to reproduce the bug?

https://github.com/ant-design/ant-design/blob/master/.stylelintrc.json

### How did you run Stylelint?

https://github.com/ant-design/ant-design/blob/764069b202e0da383569bbbea493561d6a64a462/package.json#L80

### Which version of Stylelint are you using?

14.8.4

### What did you expect to happen?

No errors like https://github.com/ant-design/ant-design/runs/6571821122?check_suite_focus=true

### What actually happened?

See errors in github action: https://github.com/ant-design/ant-design/runs/6574589909?check_suite_focus=true

<img width="772" alt="图片" src="https://user-images.githubusercontent.com/507615/170068165-c7b46813-c765-4d95-b335-c6501395ca7b.png">

### Proposal to fix the bug

revert https://github.com/stylelint/stylelint/pull/6106

**Comment by JounQin** [comment]
I can reproduce it with

```js
testRule({
  ruleName,
  customSyntax: 'postcss-less',
  config: [true],

  accept: [
    {
      code: `@alert-prefix-cls: ~'@{ant-prefix}-alert';

.@{alert-prefix-cls} {
	&-success {
    .@{alert-prefix-cls}-icon {
      color: @alert-success-icon-color;
    }
  }
}`,
      description: 'Non standard Less interpolation',
    },
  ],
})
```

Working on a fix.

**Comment by ybiquitous** [comment]
@afc163 Thanks for reporting the issue and using the template.

@JounQin Thanks for the investigation.

I can reproduce the false positive in the `no-duplicate-selectors` rule for Less syntax on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0ABYBDANjATgFwFoAHHGAMwEsAPAsTAZwF8ACYAHQDtnnUNt9ipSjTpMCFSF3ZduzSJgg54zUlADcnbo06MQAGhD0AnhzzoqCENnr194CB0oBzS9O5sQOAK7WPyt7IeHBAEUF5EmBLoeDAE9DDYYHiK9H7MeN4wmszaHLqMQA).

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jgerigmeyer** [comment]
I'm seeing the same error with Sass as well:

```scss
a {
  #{config.$link} {
    color: var(--action);
    text-decoration: none;
  }

  #{config.$focus} {
    color: var(--action-hover);
  }
}
```

```
 10:3  ✖  Unexpected duplicate selector "", first used at line 5  no-duplicate-selectors
```

**Comment by ybiquitous** [comment]
@jgerigmeyer I think PR #6111 should fix the error with Sass. Can you try the head version of Stylelint, please?

```shell
npm i github:stylelint/stylelint
```

**Comment by afc163** [comment]
Still have false alerts in 14.8.5: https://github.com/ant-design/ant-design/runs/6591263008?check_suite_focus=true

<img width="1224" alt="图片" src="https://user-images.githubusercontent.com/507615/170268256-25a83bc7-55a2-43b1-a0ca-fde5181b4f57.png">

**Comment by JounQin** [comment]
@ybiquitous The 3rd PR #6118, hope it is the last one related... 😂

**Comment by jgerigmeyer** [comment]
> @jgerigmeyer I think PR #6111 should fix the error with Sass. Can you try the head version of Stylelint, please?

v14.8.5 does fix my Sass errors as well 👍

**Comment by ybiquitous** [comment]
@afc163 Thanks for the report.

@JounQin Thanks for the investigation.

We can reproduce such false positives on the demo of both Less and SCSS syntaxes:

- [Less](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjANCgHQACwAbkgE4C+aKA9AFRGtMMo1Y673OtF2nLDRD4QAZwCemAC5IAHghAAbGBIljwETADMAlgHNlGbCnQgqAVzUSL8NNxzmQmCAFooVgA4r9YJFkYdwkYNTBZCCo7BBRZaxgnLkxRGiA)
    ```less
    a {
      b, .@{var} { /*...*/ }
      b { /*...*/ }
    }
    ```
- [SCSS](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjANCgHQDEwAJAG5IBOAvmigPQBUR7LTKdWOujrdkU7csdEPhABnAJ6YALkgAeCaWClSJ4CJgBmASwDmqjNhToQNAK4AbGFIvw0vHOZCYIAWihWADjf0wJHkYTykYOzB5CBoHBBR5axgXHkxxOiA)
    ```scss
    a {
      b, .#{$var} { /*...*/ }
      b { /*...*/ }
    }
    ```

Both emits the same message below:

```
Unexpected duplicate selector "a b", first used at line 2
```

**Comment by JounQin** [comment]
@ybiquitous It seems my PR #6118 won't fix these cases, any idea how to fix them?
